### ISSUE TITLE
Can't build with libxml 2.10.4 on Intel Mac

### ISSUE THREAD

**Issue by queue-tip** [xml4r/libxml-ruby#199]
Hello,

I've been having issues with libxml-ruby 4.0.0 and libxml 2.10.4 installed from homebrew when compiling ruby_xml_encoding.c

TIA for any pointers you can give me.

```bash
$ gem install libxml-ruby -- --with-xml2-dir=/usr/local/opt/libxml2/bin/xml2-config
Building native extensions with: '--with-xml2-dir=/usr/local/opt/libxml2/bin/xml2-config'
This could take a while...
ERROR:  Error installing libxml-ruby:
        ERROR: Failed to build gem native extension.

    current directory: /Users/user1/.asdf/installs/ruby/3.0.2/lib/ruby/gems/3.0.0/gems/libxml-ruby-4.0.0/ext/libxml
/Users/user1/.asdf/installs/ruby/3.0.2/bin/ruby -I /Users/user1/.asdf/installs/ruby/3.0.2/lib/ruby/3.0.0 -r ./siteconf20230425-54279-mints9.rb extconf.rb --with-xml2-dir\=/usr/local/opt/libxml2/bin/xml2-config
/Users/user1/.asdf/installs/ruby/3.0.2/bin/ruby: warning: shebang line ending with \r may cause problems
checking for libxml/xmlversion.h in /opt/include/libxml2,/opt/local/include/libxml2,/usr/local/include/libxml2,/usr/include/libxml2,/usr/local/include... yes
checking for xmlParseDoc() in -lxml2... yes
creating extconf.h
creating Makefile

current directory: /Users/user1/.asdf/installs/ruby/3.0.2/lib/ruby/gems/3.0.0/gems/libxml-ruby-4.0.0/ext/libxml
make DESTDIR\= clean

current directory: /Users/user1/.asdf/installs/ruby/3.0.2/lib/ruby/gems/3.0.0/gems/libxml-ruby-4.0.0/ext/libxml
make DESTDIR\=
compiling libxml.c
compiling ruby_xml.c
compiling ruby_xml_attr.c
compiling ruby_xml_attr_decl.c
compiling ruby_xml_attributes.c
compiling ruby_xml_cbg.c
compiling ruby_xml_document.c
compiling ruby_xml_dtd.c
compiling ruby_xml_encoding.c
ruby_xml_encoding.c:200:3: error: incompatible function pointer types passing 'VALUE (VALUE, VALUE)' (aka 'unsigned long (unsigned long, unsigned long)') to parameter of type 'VALUE (*)(VALUE, VALUE, VALUE)' (aka 'unsigned long (*)(unsigned long, unsigned long, unsigned long)') [-Wincompatible-function-pointer-types]
  rb_define_module_function(mXMLEncoding, "to_rb_encoding", rxml_encoding_to_rb_encoding, 2);
  ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/Users/user1/.asdf/installs/ruby/3.0.2/include/ruby-3.0.0/ruby/internal/anyargs.h:337:142: note: expanded from macro 'rb_define_module_function'
#define rb_define_module_function(mod, mid, func, arity)    RBIMPL_ANYARGS_DISPATCH_rb_define_module_function((arity), (func))((mod), (mid), (func), (arity))
                                                                                                                                             ^~~~~~
/Users/user1/.asdf/installs/ruby/3.0.2/include/ruby-3.0.0/ruby/internal/anyargs.h:273:1: note: passing argument to parameter here
RBIMPL_ANYARGS_DECL(rb_define_module_function, VALUE, const char *)
^
/Users/user1/.asdf/installs/ruby/3.0.2/include/ruby-3.0.0/ruby/internal/anyargs.h:256:72: note: expanded from macro 'RBIMPL_ANYARGS_DECL'
RBIMPL_ANYARGS_ATTRSET(sym) static void sym ## _02(__VA_ARGS__, VALUE(*)(VALUE, VALUE, VALUE), int); \
                                                                       ^
1 error generated.
make: *** [ruby_xml_encoding.o] Error 1

make failed, exit code 2

Gem files will remain installed in /Users/user1/.asdf/installs/ruby/3.0.2/lib/ruby/gems/3.0.0/gems/libxml-ruby-4.0.0 for inspection.
Results logged to /Users/user1/.asdf/installs/ruby/3.0.2/lib/ruby/gems/3.0.0/extensions/x86_64-darwin-22/3.0.0/libxml-ruby-4.0.0/gem_make.out
```

**Comment by cfis** [comment]
Thanks for the report!

**Comment by dmolesUC** [comment]
IDK if the 3.2.x line is still supported, but I've got some legacy dependencies that require it, and FWIW I was able to apply the fix from 1d51054 to 3.2.4: https://github.com/dmolesUC/libxml-ruby/tree/macos-3.2.x-fix
